### ISSUE TITLE
Bug: "List and sort directories alongside files" only works in Columns view

### ISSUE THREAD

**Issue by luigman** [files-community/Files#12177]
### Description

When I enable "List and sort directories alongside files", the app does not apply this setting to any view other than Columns. For example, if I select the Details or Icons view, then files and directories are still sorted separately. This setting should be applied to all views so it is consistent.

### Steps To Reproduce

1. Enable "List and sort directories alongside files" in settings
2. Open a directory in Details view. The setting is not applied.

### Requirements

- Sort directories alongside files in all view modes.

### Files Version

2.4.67.0

### Windows Version

Windows 10 Education 22H2 19045.2846 Windows Feature Experience Pack 120.2212.4190.0

### Log file

[debug - Copy.log](https://github.com/files-community/Files/files/11304945/debug.-.Copy.log)

**Comment by yaira2** [comment]
@luigman do you have the option to sync folder settings turned on?
![image](https://user-images.githubusercontent.com/39923744/233866359-11de2ecd-b9e6-45c1-bde4-d6ec6ba198b6.png)

**Comment by luigman** [comment]
Yes, I have that option enabled. The issue is that it is only applied when I'm in the "Columns" view. Here's some example images of what it looks like in Columns vs. Details view.

![Screenshot 2023-04-23 175901](https://user-images.githubusercontent.com/4612263/233877544-62bf851e-3748-4ff9-89d1-de975d778b79.png)
![Screenshot 2023-04-23 175958](https://user-images.githubusercontent.com/4612263/233877549-922d92ac-23a4-4af8-a290-13e111b02921.png)

**Comment by yaira2** [comment]
I'm having a hard time reproducing this issue, if anyone is having this issue, please let us know.

**Comment by yaira2** [comment]
This might be related to #12045

**Comment by yaira2** [comment]
@luigman are you still able to repro this in the latest preview version? https://files.community/download

**Comment by luigman** [comment]
I just tried it out, and this is still an issue in the latest preview

**Comment by yaira2** [comment]
Can you share a screenshot of the full window so I can get a better look?

**Comment by luigman** [comment]
Sure, here's the comparison of columns and details view.

![Screenshot 2023-10-14 113058](https://github.com/files-community/Files/assets/4612263/929f5934-8de6-403f-b3b5-1a382ba34ffa)
![Screenshot 2023-10-14 113118](https://github.com/files-community/Files/assets/4612263/5551b3ad-e41c-4764-bcb4-e6863758934d)

**Comment by yaira2** [comment]
Do you have the group by option turned on?

**Comment by luigman** [comment]
No, group-by is turned off (set to None).

I finally figured this out though! There are two "List and sort directories alongside files" settings. I had enabled the one in settings:
![Screenshot 2023-10-16 210212](https://github.com/files-community/Files/assets/4612263/9db653f7-fbd1-4810-943f-6b9b852e68bd)

But there is another option in the top-right, which was disabled when I was using details view.
![Screenshot 2023-10-16 210431](https://github.com/files-community/Files/assets/4612263/012d603a-4359-463c-b514-1d198128adf1)

I'm still not super clear on what the option in Settings is supposed to do. If "Sync layout and sorting preferences across directories" is enabled then it's clear that it's a global setting, but otherwise it seems like this gets out of sync with the option for the current directory.
